src/main/scala/temple/builder/DatabaseBuilder.scala

```diff
@@ -0,0 +1,57 @@
+package temple.builder
+
+import temple.DSL.semantics.{Annotation, Attribute, AttributeType, ServiceBlock}
+import temple.generate.database.ast.ColumnConstraint.Check
+import temple.generate.database.ast._
+
+/** Construct database queries from a Templefile structure */
+object DatabaseBuilder {
+
+  private def generateMaxMinConstraints[T](name: String, max: Option[T], min: Option[T]): Seq[ColumnConstraint] = {
```

Comment:
`[T: Numeric]` may be useful as the type parameter

Reply:
Don’t think that’s exactly true - we wouldn’t gain anything from it, and in the future we wouldn’t be able to produce constraints over two variables

```
CHECK val_a >= val_b
```